### DISCUSSION TITLE
Get Active Branch

### DISCUSSION THREAD

**Discussion by dufferzafar** [jelmer/dulwich#945] [Q&A] [answered]
There is `dulwich.porcelain.branch_list()` but that will list all branches for a repo.

How does one get the active branch? (the one displayed when you run `git status`)

**Comment by jelmer** [comment, marked as answer]
Try something like:

     >>> from dulwich.repo import Repo
     >>> Repo('.').refs.follow(b'HEAD')
     (['HEAD', 'refs/heads/master'], '4349a6e7e86d3566678657dfe4719b8a026293d9')

or more specifically:

    >>> Repo('.').refs.follow(b'HEAD')[0][-1].lstrip(b'refs/heads/')
    'master'

**Reply by jelmer** [reply]
With modern versions of dulwich, there is now also:

```
>>> from dulwich import porcelain
>>> porcelain.active_branch('.')
b'master'
```
